lib/LaTeXML/Core/State.pm

```diff
@@ -227,6 +227,23 @@ sub shiftValue {
   assign_internal($self, 'value', $key, [], 'global') unless $$vtable{$key}[0];
   return shift(@{ $$vtable{$key}[0] }); }
 
+sub removeValue {
+  my ($self, $key) = @_;
+  if (my $vvec = $$self{value}{$key}) {
+    my $value;
+    # undo ONLY the most recent binding for this value.
+    my $frame;
+    my @frames = @{ $$self{undo} };
+    while (@frames) {
+      $frame = shift(@frames);
+      if (my $n = $$frame{value}{$key}) {    # Undo the bindings, if $key was bound in this frame
```

Comment:
Is this idempotent? I.e. if applied twice, will it remove the *two* previous bindings? That would be too much for `\box`.

Reply:
Curious idea - we don't yet have a test for that. Something like:

```
\newbox\myboxD
\setbox\myboxD\vbox{D won't appear outer.}
\begingroup
  \setbox\myboxD\vbox{D will appear inner.}
  \box\myboxD
  \box\myboxD
\endgroup
\bye
```

Reply:
Added a test. It appears to work as expected?

Reply:
Scratch that, the test wasn't precise enough -- there is indeed a bug where unbinding twice will pop the outer frame, nice catch @xworld21

Reply:
Ok, using `exists` and a `0` value for the undo counter becomes idempotent. The new test now has correct timing and matches the PDF.

But the code got extra scary. Maybe good, maybe not?